### PR TITLE
MINOR: Add unit test for SerDe auto-configuration

### PR DESCRIPTION
**Pull request by mjsax**
Follow up of (including) #6461 (first commit only -- original commits squashed)

If we are happy with this, we can merge #6461 and than I can rebase this and drop the first commit.

### CODE REVIEW

**Comment by mjsax** [comment]
Call for review @tedyu @guozhangwang @bbejeck @vvcephei @ableegoldman @cadonna

File: streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java [hunk 1]
```diff
@@ -661,6 +673,131 @@ public void statefulTopologyShouldCreateStateDirectory() throws Exception {
         startStreamsAndCheckDirExists(topology, asList(inputTopic, globalTopicName), outputTopic, true);
     }
 
+    @SuppressWarnings("unchecked")
+    @Test
+    public void shouldInitializedUserSerdes() {
```
Comment (tedyu) [inline streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java:678]: The verb following 'should' would be in active tense.

consider naming this test: userSerdesShouldBeInitialized

File: streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java [hunk 2]
```diff
@@ -661,6 +673,131 @@ public void statefulTopologyShouldCreateStateDirectory() throws Exception {
...
+        final Deserializer mockThroughValueDeserializer = mock(Deserializer.class);
+        mockThroughValueDeserializer.configure(anyObject(), eq(false));
+        expectLastCall();
+
+        final Serde mockThroughKeySerde = mock(Serde.class);
+        final Serde mockThroughValueSerde = mock(Serde.class);
+        expect(mockThroughKeySerde.serializer()).andReturn(mockThroughKeySerializer).anyTimes();
+        expect(mockThroughValueSerde.serializer()).andReturn(mockThroughValueSerializer).anyTimes();
+        expect(mockThroughKeySerde.deserializer()).andReturn(mockThroughKeyDeserializer).anyTimes();
+        expect(mockThroughValueSerde.deserializer()).andReturn(mockThroughValueDeserializer).anyTimes();
```
Comment (tedyu) [inline streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java:711]: should this expect be aligned with the above 'expect's ?

File: streams/src/test/java/org/apache/kafka/streams/state/internals/MeteredKeyValueStoreTest.java [hunk 3]
```diff
@@ -101,6 +103,53 @@ private void init() {
         metered.init(context, metered);
     }
 
+    @SuppressWarnings("unchecked")
+    @Test
+    public void shouldGetSerdesFromConfigIfNoUserSerdes() {
```
Comment (tedyu) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/MeteredKeyValueStoreTest.java:108]: A verb is normally expected following 'if no xx'
How about naming this test:

shouldGetSerdesFromConfigWithouUserSerdes

**Review by tedyu — COMMENTED** [review]
Overall the tests look good.

**Comment by mjsax** [comment]
Pushed this branch to wrong repo. Replaced with #6610

Will address your comment on the new PR @tedyu -- thanks for reviewing!

File: streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java (hunk 2 above)
Comment (mjsax) [reply streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java:711]: What do you mean exactly? What lines are you referring to?

Comment (tedyu) [reply streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java:711]: In the current PR, these are aligned.
Dropping former comment.

**Pull request closed without merging**